projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php

```diff
@@ -71,8 +74,21 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
 		}
 
 		if ( $is_valid_token ) {
-			if ( 'subscribers' === $access_level ) {
-				return 'active' === $payload['blog_sub'];
+			if ( self::POST_ACCESS_LEVEL_SUBSCRIBERS === $access_level ) {
```

Comment:
I have modified all that logic in [this PR](https://github.com/Automattic/jetpack/pull/27352). Do you mind to wait and rebase on it? It should be minimal work compared to the other way around ....